code/cycler/src/wattrex_battery_cycler/mid/mid_str/mid_str_node.py

```diff
@@ -0,0 +1,166 @@
+### THREAD ###
+class MidStrNodeC(SysShdNodeC): #pylint: disable= too-many-instance-attributes
+    """This class will create a node that communicates with the databases reading and writing data.
+    """
+    def __init__(self, name: str, working_flag : Event, shared_gen_meas: SysShdSharedObjC, #pylint: disable= too-many-arguments
+                 shared_ext_meas: SysShdSharedObjC, shared_status: SysShdSharedObjC,
+                 str_reqs: SysShdChanC, str_data: SysShdChanC, str_alarms: SysShdChanC,
+                 cycle_period: int, cycler_station: int, master_file: str, cache_file: str,
+                 str_params: SysShdNodeParamsC= SysShdNodeParamsC()) -> None:
+        '''
+        Initialize the MID_STR thread used as database proxy.
+
+        Args:
+            cycle_period (int): Period of the thread cycle in seconds.
+            working_flag (threading.Event): Flag used to stop the thread.
+            name (str, optional): Name of the thread. Defaults to 'MID_STR'.
+        '''
+        super().__init__(name, cycle_period, working_flag, str_params)
+        log.info(f"Initializing {name} node...")
+        self.db_iface = MidStrFacadeC(master_file= master_file, cache_file= cache_file,
+                                      cycler_station_id= cycler_station)
+        self.str_reqs: SysShdChanC = str_reqs
+        self.str_data: SysShdChanC = str_data
+        self.str_alarms: SysShdChanC = str_alarms
+        self.globlal_gen_meas: SysShdSharedObjC = shared_gen_meas
+        self.globlal_ext_meas: SysShdSharedObjC = shared_ext_meas
+        self.globlal_all_status: SysShdSharedObjC = shared_status
+        self.__actual_exp_id: int = -1
+        self.__new_raised_alarms: List[CyclerDataAlarmC] = []
+        ## Once it has been initilizated all atributes ask for the cycler station info
+        cycler_info = self.db_iface.get_cycler_station_info()
+        self.str_data.send_data(MidStrCmdDataC(cmd_type= MidStrDataCmdE.CS_DATA,
+                                               station= cycler_info))
+
+    def __receive_alarms(self) -> None:
+        alarm = self.str_alarms.receive_data_unblocking()
+        while alarm is not None:
+            self.__new_raised_alarms.append(alarm)
+            alarm = self.str_alarms.receive_data_unblocking()
+
+    def __apply_command(self, command : MidStrCmdDataC) -> None:
+        '''Apply a command to the Storage node.
+
+        Args:
+            command (DrvCanCmdDataC): Data to process, does not know if its for the channel or
+            a message to a device
+
+        Raises:
+            err (CanOperationError): Raised when error with CAN connection occurred
+        '''
+        #Check which type of command has been received and matchs the payload type
+        if command.cmd_type == MidStrReqCmdE.GET_NEW_EXP:
+            log.info('Getting new experiment info from database')
+            exp_info, battery_info, profile_info = self.db_iface.get_start_queued_exp()
```

Comment:
What happens if not experiment has been found?

Reply:
Manager will receive an error flag in the message and handle it.